src/sysinfo.cc

```diff
@@ -706,8 +706,8 @@ std::vector<double> GetLoadAvg() {
 }  // end namespace
 
 const CPUInfo& CPUInfo::Get() {
-  static const CPUInfo* info = new CPUInfo();
-  return *info;
+  static const CPUInfo info;
```

Comment:
this is not safe as it has a non-trivial destructor. The destructor will be executed by the thread which starts the termination of the program, but other threads may not have terminated yet and _might_ still be referencing it.

The only safe way to manage this is having safe initialization with no destruction. Either declaring things as `constexpr`, or heap allocating the static object.

Reply:
Yes, I agree that that destruction can become a problem with this version of the changes. However, if we can find a solution that does not trip valgrind into thinking there is a memory leak, that would be better. I don't think using `constexpr` is possible, because `CPUInfo` queries information about the hardware at runtime. Maybe a call to delete the allocated object can be registered with `atexit()`?

Reply:
that's pretty much how the global destructors run now (hand waving around a bit) and it suffers from the same issue.

we just don't know what people are doing with the library and threads that could cause issues.

having a way to tell valgrind "this is not a leak" would be great. with LSAN we can pass in a [leak suppression file](https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer) but i don't think valgrind has something similar, does it?

Reply:
It does, you can pass a leak suppression file. I guess that if there's no easy way to silence valgrind without causing other problems, then I will just apply this change locally within my project, as I always run the benchmarks with a single thread there and this change is not an issue.